### ISSUE TITLE
[JENKINS-6846] Poll SCM with post-commit hook or by cron-definition

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#8319]
<p>We have splitted some projects from one single job (doing build/test/analyse) to two jobs (one only for build and second one for test/analyse). The first job polls scm every 10 minutes, the second one polls scm at midnight.</p>

<p>Now we defined a svn post-commit hook for CI, so we don't have to poll every 10 minutes for changes. This is fine for the first job, but the second one now starts too (because 'Poll SCM' is checked). For now it is not possible to define a job, that polls scm due a cron-definition and another one that is triggered by post-commit hook.</p>

<p>It would be nice to have an additional option like "ignore post-commit hook". So 'Poll SCM' could be checked and triggered by cron-definition and not by post-commit.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=manthos">manthos</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-6846" target="_blank">Poll SCM with post-commit hook or by cron-definition</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kutzi.jpg" title="kutzi's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kutzi">kutzi</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core, subversion-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2012-11-24T15:25:00+00:00
<li><b>votes</b>: 5
<li><b>watchers</b>: 9
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>We have splitted some projects from one single job (doing build/test/analyse) to two jobs (one only for build and second one for test/analyse). The first job polls scm every 10 minutes, the second one polls scm at midnight.</p>

<p>Now we defined a svn post-commit hook for CI, so we don't have to poll every 10 minutes for changes. This is fine for the first job, but the second one now starts too (because 'Poll SCM' is checked). For now it is not possible to define a job, that polls scm due a cron-definition and another one that is triggered by post-commit hook.</p>

<p>It would be nice to have an additional option like "ignore post-commit hook". So 'Poll SCM' could be checked and triggered by cron-definition and not by post-commit.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gbois:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-6846?focusedId=151719&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-151719">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>@manthos
request information:
1) Do you have think that the second job could poll build result of job1 instead of SCM
2) How do you schedule the first job in the post commit hook?</p>
</pre>
</details>

</ul>
</details>

<p>@​manthos<br/>
request information:<br/>
1) Do you have think that the second job could poll build result of job1 instead of SCM<br/>
2) How do you schedule the first job in the post commit hook?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>manthos:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-6846?focusedId=151726&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-151726">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>1) this doesn't change anything. if I bind job2 to job1, job2 builds right after job1...
2) first job starts building by a scm postcommithook (push), no cronjob needed.</p>

<p>IMO 'Poll SCM' and 'postcommithook (push)' are different triggers. Currently you are not able to have different Builds (based on same scm repository) but one triggered by postcommithook (push), the other by poll.</p>

<p>Example:
Job1:</p>
<ul class="alternate" type="square">
	<li>maven deploy -DskipTests=true of <a href="http://svn.foo.bar/svn/java/foobarproject" class="external-link" target="_blank" rel="nofollow noopener">http://svn.foo.bar/svn/java/foobarproject</a></li>
	<li>build triggered by scm postcommithook (push)
So every time a user checks something into scm, foobarproject builds, skipping the (time consuming) tests</li>
</ul>


<p>Job2:</p>
<ul class="alternate" type="square">
	<li>maven test of <a href="http://svn.foo.bar/svn/java/foobarproject" class="external-link" target="_blank" rel="nofollow noopener">http://svn.foo.bar/svn/java/foobarproject</a></li>
	<li>build triggered by Poll SCM, in ex: (Mo-Fr at 23:00 -&gt; 0 23 * * 1,2,3,4,5)
So at 23:00 -&gt; tests of foobarproject but only if something checked in (since last run)</li>
</ul>


</pre>
</details>

</ul>
</details>

<p>1) this doesn't change anything. if I bind job2 to job1, job2 builds right after job1...<br/>
2) first job starts building by a scm postcommithook (push), no cronjob needed.</p>

<p>IMO 'Poll SCM' and 'postcommithook (push)' are different triggers. Currently you are not able to have different Builds (based on same scm repository) but one triggered by postcommithook (push), the other by poll.</p>

<p>Example:<br/>
Job1:</p>
<ul class="alternate" type="square">
	<li>maven deploy -DskipTests=true of <a href="http://svn.foo.bar/svn/java/foobarproject" class="external-link" target="_blank" rel="nofollow noopener">http://svn.foo.bar/svn/java/foobarproject</a></li>
	<li>build triggered by scm postcommithook (push)<br/>
So every time a user checks something into scm, foobarproject builds, skipping the (time consuming) tests</li>
</ul>


<p>Job2:</p>
<ul class="alternate" type="square">
	<li>maven test of <a href="http://svn.foo.bar/svn/java/foobarproject" class="external-link" target="_blank" rel="nofollow noopener">http://svn.foo.bar/svn/java/foobarproject</a></li>
	<li>build triggered by Poll SCM, in ex: (Mo-Fr at 23:00 -> 0 23 * * 1,2,3,4,5)<br/>
So at 23:00 -> tests of foobarproject but only if something checked in (since last run)</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rainerw:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-6846?focusedId=161519&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-161519">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>+1 
Same Problem: running nightly UI tests, which take about 3-4 hours to finish. I want this to run only after an actual change, because running every midnight will make the test-result-history kind of useless.</p>

<p>Therefore this is definitively a critical bug!</p>
</pre>
</details>

</ul>
</details>

<p>+1 <br/>
Same Problem: running nightly UI tests, which take about 3-4 hours to finish. I want this to run only after an actual change, because running every midnight will make the test-result-history kind of useless.</p>

<p>Therefore this is definitively a critical bug!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-6846?focusedId=170323&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-170323">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Christoph Kutzinski
Path:
 changelog.html
 core/src/main/java/hudson/triggers/SCMTrigger.java
 core/src/main/resources/hudson/triggers/SCMTrigger/config.jelly
 core/src/main/resources/hudson/triggers/SCMTrigger/help-ignorePostCommitHooks.html
<a href="http://jenkins-ci.org/commit/jenkins/b65c30cb368e79205edacd958d8fd98dd89f3a73" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/b65c30cb368e79205edacd958d8fd98dd89f3a73</a>
Log:
  Switch to ignore post-commit hook in SCM polling triggers <span class="error">&#91;FIXED JENKINS-6846&#93;</span></p>




</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Christoph Kutzinski<br/>
Path:<br/>
 changelog.html<br/>
 core/src/main/java/hudson/triggers/SCMTrigger.java<br/>
 core/src/main/resources/hudson/triggers/SCMTrigger/config.jelly<br/>
 core/src/main/resources/hudson/triggers/SCMTrigger/help-ignorePostCommitHooks.html<br/>
<a href="http://jenkins-ci.org/commit/jenkins/b65c30cb368e79205edacd958d8fd98dd89f3a73" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/b65c30cb368e79205edacd958d8fd98dd89f3a73</a><br/>
Log:<br/>
  Switch to ignore post-commit hook in SCM polling triggers <span class="error">&#91;FIXED JENKINS-6846&#93;</span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-6846?focusedId=170325&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-170325">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/2096/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #2096</a>
     Switch to ignore post-commit hook in SCM polling triggers <span class="error">&#91;FIXED JENKINS-6846&#93;</span> (Revision b65c30cb368e79205edacd958d8fd98dd89f3a73)</p>

<p>     Result = UNSTABLE
Christoph Kutzinski : <a href="https://github.com/jenkinsci/jenkins/commit/b65c30cb368e79205edacd958d8fd98dd89f3a73" class="external-link" target="_blank" rel="nofollow noopener">b65c30cb368e79205edacd958d8fd98dd89f3a73</a>
Files : </p>
<ul>
	<li>core/src/main/resources/hudson/triggers/SCMTrigger/help-ignorePostCommitHooks.html</li>
	<li>changelog.html</li>
	<li>core/src/main/resources/hudson/triggers/SCMTrigger/config.jelly</li>
	<li>core/src/main/java/hudson/triggers/SCMTrigger.java</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/2096/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #2096</a><br/>
     Switch to ignore post-commit hook in SCM polling triggers <span class="error">&#91;FIXED JENKINS-6846&#93;</span> (Revision b65c30cb368e79205edacd958d8fd98dd89f3a73)</p>

<p>     Result = UNSTABLE<br/>
Christoph Kutzinski : <a href="https://github.com/jenkinsci/jenkins/commit/b65c30cb368e79205edacd958d8fd98dd89f3a73" class="external-link" target="_blank" rel="nofollow noopener">b65c30cb368e79205edacd958d8fd98dd89f3a73</a><br/>
Files : </p>
<ul>
	<li>core/src/main/resources/hudson/triggers/SCMTrigger/help-ignorePostCommitHooks.html</li>
	<li>changelog.html</li>
	<li>core/src/main/resources/hudson/triggers/SCMTrigger/config.jelly</li>
	<li>core/src/main/java/hudson/triggers/SCMTrigger.java</li>
</ul>
